service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunSetsPoller.java

```diff
@@ -99,6 +102,30 @@ private RunSet updateRunSet(RunSet rs, OffsetDateTime limitedEndTime) {
       smartRunsPoller.updateRuns(updateableRuns, Optional.of(runPollUpdateEndTime));
 
       StatusAndCounts newStatusAndCounts = newStatusAndErrorCounts(rs);
+
+      if (rs.status() == CbasRunSetStatus.CANCELING) {
+        abortManager.abortRunSet(rs.runSetId());
+        List<Run> allRuns = runDao.getRuns(new RunDao.RunsFilters(rs.runSetId(), null));
+        int canceledRuns = 0;
+
+        // Check how many runs in the run set are canceled; increment.
+        for (Run run : allRuns) {
+          if (run.status() == CbasRunStatus.CANCELED) {
+            canceledRuns += 1;
+          }
+        }
```

Comment:
Can this information be fetched from the database itself? I think the `getRunStatusCounts` can be used to filter by RunSet Id and Run status.